### PR TITLE
Add pkg-config to header detect and version bump math lib

### PR DESCRIPTION
**Pull request by jopadan**

### CODE REVIEW

File: depends/check-dependencies.sh [hunk 1]
```diff
@@ -54,7 +80,7 @@ if [ "$(uname)" == "Darwin" ]; then
 fi
 
 check_header    libelf          elf.h libelf.h libelf/libelf.h gelf.h libelf/gelf.h
-check_header    libusb          usb.h
+check_header    libusb          libusb.h
```
Comment (sharkwouter) [inline depends/check-dependencies.sh:83]: This is probably not a good idea. We need either libusb 1 or libusb-compat, which both ship ``usb.h``. The code will not build with newer libusb versions.

Comment (jopadan) [reply depends/check-dependencies.sh]: I changed it to check for `usb.h` header and then for `libusb.h` both advanced with pkg-config.
It returns the first matched of the header names listed in `check_header libusb usb.h libusb.h`.

Comment (sharkwouter) [reply depends/check-dependencies.sh]: Which program is using libusb.h to build?

Comment (jopadan) [reply depends/check-dependencies.sh]: gentoo version of libusb-1.0.24-r1 uses libusb-1.0/libusb.h
gentoo version of libusb-compat-0.1.5-r3 uses libusb-compat/usb.h
the psp dependency is psplinkusb and probably more.

**Pull request closed without merging**